### ISSUE TITLE
Suggestion by unnecessary_unwrap reverses conditions 

### ISSUE THREAD

**Issue by Gowee** [rust-lang/rust-clippy#11006]
### Summary

As in the title.

### Reproducer

I tried this code:

```rust
fn main() {
     let a = Some(1);
     let _b = if let Some(..) = a {
         println!("Hello, world!");
         99
     } else{
         a.unwrap()
     };
 }
```

I expected to see this happen:
N/A

Instead, this happened:
```
warning: called `unwrap` on `a` after checking its variant with `is_none`
 --> src/lib.rs:7:9
  |
3 |     let _b = if a.is_none() {
  |              -------------- help: try: `if let Some(..) = a`
...
7 |         a.unwrap()
  |         ^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_unwrap
  = note: `#[warn(clippy::unnecessary_unwrap)]` on by default
```

### Version

```text
rustc 1.71.0-nightly (18bfe5d8a 2023-05-14)
binary: rustc
commit-hash: 18bfe5d8a9ca0e226171e98f8f4ef071790f3352
commit-date: 2023-05-14
host: x86_64-unknown-linux-gnu
release: 1.71.0-nightly
LLVM version: 16.0.2
```

### Additional Labels

@rustbot label +I-suggestion-causes-error

**Comment by Centri3** [comment]
Also looks like an issue with `is_err`.
@rustbot claim
